homeassistant/components/integration/sensor.py

```diff
@@ -115,16 +124,28 @@ def __init__(
 
         self._unit_prefix = UNIT_PREFIXES[unit_prefix]
         self._unit_time = UNIT_TIME[unit_time]
+        self._attr_state_class = STATE_CLASS_MEASUREMENT
 
     async def async_added_to_hass(self):
         """Handle entity which will be added."""
         await super().async_added_to_hass()
         state = await self.async_get_last_state()
+        self._attr_last_reset = dt_util.utcnow()
         if state:
             try:
                 self._state = Decimal(state.state)
             except ValueError as err:
                 _LOGGER.warning("Could not restore last state: %s", err)
+            else:
+                if state.attributes.get(ATTR_LAST_RESET) is None:
+                    self._attr_last_reset = dt_util.utc_from_timestamp(0)
+                else:
+                    last_reset = dt_util.parse_datetime(
+                        state.attributes[ATTR_LAST_RESET]
+                    )
+                    self._attr_last_reset = (
+                        last_reset if last_reset else dt_util.utc_from_timestamp(0)
+                    )
```

Comment:
Not 100% sure, but I guess you could do this:

```suggestion
                last_reset = dt_util.parse_datetime(
                    state.attributes.get(ATTR_LAST_RESET, "")
                )
                self._attr_last_reset = (
                    last_reset if last_reset else dt_util.utc_from_timestamp(0)
                )
```

Although bit less explicit maybe, so maybe not... feel free to ignore :)

Reply:
I like it :)